packages/cli/src/bundle/metro.ts

```diff
@@ -58,17 +38,44 @@ export async function metroBundle(
     sourceMapPath = path.join(distPath, sourceMapPath);
   }
 
+  return {
+    assetsDest: bundleConfig.assetsPath,
+    entryFile: bundleConfig.entryPath,
+    minify: bundleConfig.minify,
+    platform,
+    dev: bundleConfig.dev,
+    bundleOutput: bundlePath,
+    bundleEncoding: bundleConfig.bundleEncoding,
+    sourcemapOutput: sourceMapPath,
+    sourcemapSourcesRoot: bundleConfig.sourceMapSourceRootPath,
+  };
+}
+
+/**
+ * Run the Metro bundler.
+ *
+ * @param tsservice TypeScript service to use for type-checking (when enabled)
+ * @param metroConfig Metro configuration
+ * @param bundleConfig Bundle configuration
+ */
+export async function metroBundle(
+  tsservice: Service,
+  metroConfig: ConfigT,
+  bundleConfig: BundleConfig
```

Comment:
Nit: We can also extract everything we need from `BundleConfig` here:
```suggestion
  {
    detectCyclicDependencies,
    detectDuplicateDependencies,
    dev,
    entryPath,
    experimental_treeShake,
    minify
    platform,
    typescriptValidation,
  }: BundleConfig
```